src/main/java/codeu/controller/ConversationServlet.java

```diff
@@ -116,6 +118,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     Conversation conversation =
         new Conversation(UUID.randomUUID(), user.getId(), conversationTitle, Instant.now());
 
+    // Get all tags
+    String tags = request.getParameter("tags");
+    if (tags != null) {
+      if (!tags.matches("([\\w*](, ))*[\\w*]")) {
+        request.setAttribute("error", "Please enter tags as comma-separated words with one space between them. Tags can only contain letters and numbers.");
+        request.getRequestDispatcher("/WEB-INF/view/conversations.jsp").forward(request, response);
+        return;
+      }
+
+      // Add all tags to the user's tagged conversations Map
+      List<String> splitTags = Arrays.asList(tags.split(", "));
+      for (String tag : splitTags) {
+        Tag newTag = new Tag(UUID.randomUUID(), tag, Instant.now());
+        user.addTaggedConversation(newTag, conversation);
```

Comment:
Changes to make:
1. Check if Tag exists in TagStore
2. Add Tag to Conversation and add Conversation to Tag.